Fix: Avoid Java bug caused by readString for readPreferencesFile

magge-faf opened the pull request:
Related to https://github.com/FAForever/downlords-faf-client/issues/2905

Commit:
Fix: Avoid Java bug caused by readString for readPreferencesFile

Sheikah45 commented on src/main/java/com/faforever/client/mod/ModService.java:
```diff
@@ -308,17 +313,26 @@ private String readPreferencesFile(Path preferencesFile) throws IOException {
+      } catch (CharacterCodingException e) {
+        log.info("Failed to read preferences file with charset: " + charset.displayName());
+        // Continue and try a different character set
+      } catch (IOException e) {
+        // User has a malformed file
+        throw new RuntimeException(e);
```
I would still somewhere just catch all exceptions just in case

Commit:
Handle all exceptions while reading preferences file

magge-faf commented:
- Added catch-all-block just in case there are unforeseen issues.
- Changed comment into log.error to clarify that is an IOException issue and not only a "probably malformed file"

It worked on all the tests files, which I got from the forum - I converted a random Game.prefs to several commonly used char sets (https://www.wikiwand.com/en/Character_encoding), and it worked for them as well - I think this alternate solution to avoid the read-decoder-Java-bug, is the next best one. 

@Sheikah45 From my POV the solution is ready, I tested all successfully I could on my end. If I find some volunteer who have the current bug, I could test the new client with them first, if you think this is not wasted time.
